snap/quota/resources_test.go

```diff
@@ -0,0 +1,62 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package quota_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/gadget/quantity"
+	"github.com/snapcore/snapd/snap/quota"
+)
+
+type resourcesTestSuite struct{}
+
+var _ = Suite(&resourcesTestSuite{})
+
+func (s *resourcesTestSuite) TestQuotaValidation(c *C) {
+	tests := []struct {
+		limits quota.Resources
+		err    string
+	}{
+		{quota.Resources{}, `quota group must have a memory limit set`},
+		{quota.NewResources(quantity.Size(0)), `quota group must have a memory limit set`},
```

Comment:
(nitpick) Maybe also a non-failing test here (and below) to ensure that we can actually call validate and it does  not always fails :) ? Fine in a followup fwiw.